### DISCUSSION TITLE
Proper way to do table update with pagination

### DISCUSSION THREAD

**Discussion by gordonchanhk** [remix-run/remix#7159] [General]
I am new to Remix but with some experience in ReactJS. I understand Remix render pages in SSR way, but I also wonder how to work out page like having a table with order records but paginated. I assume clicking the next page will fire ajax to route for next page records so then update the table.
What I read is that I should use useFetcher, and fetcher will call the route so the loader will get the data.
So far I see examples all use search param to achieve the goal, while for my case, I wanna pass the direction (next/prev) and a graphql resource cursor in form of form post.
What I experience is, since I am doing formPost, I have no issue to get the form data with `await request.formData()`, but the loader can't. I can get the wanted data from action function, but the loader still run afterward. 

Am I thinking in the wrong approach? Is there a way to skip the loader running and only let the action to return data?

**Comment by gordonchanhk** [comment]
Hi Mike,

Thanks for your prompt input. I read the pagination approach (with fews
other online tutorial) but not too fit my need. I would go for using
fetcher which result in both action and loader being called.
For loader, I read there is no form data, any hint for getting the form
fields that you mentioned? I’d like to find way to skip the loader doing
the hard works upon ajax fetching the sub pages~

Regards, Gordon

**Comment by chenxiaoyao6228** [comment]
I have the same issue when building a tree component, since the data is fetched by level(when the parent folder expand) and merge into a tree strcture on the client and it's not so simple as the doc described that we can simply derived state from the loader response
